cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/service/BasicServiceWorkerContext.java

```diff
@@ -95,10 +96,17 @@ public BasicServiceWorkerContext(ServiceWorkerSpecification spec, Program progra
     this.userMetrics = new ProgramUserMetrics(getMetricCollector(metricsSerice, program,
                                                                  spec.getName(), runId.getId(), instanceId));
     this.runtimeArgs = runtimeArgs.asMap();
+
+    // The cache expiry should be greater than (2 * transaction.timeout) and at least 2 minutes.
+    // This ensures that when a dataset instance is requested multiple times during a single transaction,
+    // the same instance is always returned.
```

Comment:
Not unless a dataset is cached in a first transaction, and then in the course of another transaction, it could get expired. Right? Or is this ``BasicServiceWorkerContext`` only meant to live for the course of one transaction only?